src/hooks/useAutoSlippageTolerance.ts

```diff
@@ -77,30 +78,31 @@ export default function useClassicAutoSlippageTolerance(trade?: ClassicTrade): P
   const { chainId } = useWeb3React()
   const onL2 = chainId && L2_CHAIN_IDS.includes(chainId)
   const outputDollarValue = useStablecoinValue(trade?.outputAmount)
-  const nativeGasPrice = useGasPrice()
 
+  const nativeGasPrice = useGasPrice()
   const gasEstimate = guesstimateGas(trade)
   const gasEstimateUSD = useStablecoinAmountFromFiatValue(trade?.gasUseEstimateUSD) ?? null
   const nativeCurrency = useNativeCurrency(chainId)
-  const nativeCurrencyPrice = useStablecoinPrice((trade && nativeCurrency) ?? undefined)
+
+  const nativeGasCost =
+    nativeGasPrice && typeof gasEstimate === 'number'
+      ? JSBI.multiply(nativeGasPrice, JSBI.BigInt(gasEstimate))
+      : undefined
+  const gasCostUSD = useUSDPrice(
+    trade && nativeCurrency && nativeGasCost ? CurrencyAmount.fromRawAmount(nativeCurrency, nativeGasCost) : undefined
+  )
+  const gasCostStablecoinAmount = useStablecoinAmountFromFiatValue(gasCostUSD.data)
```

Comment:
so dont htink you have to convert this to a stablecoin value, its actually more accurate for autoslippage to just do USD 

so the calculation is like, we want the ratio of gas to trade size, and trade size is currently in terms of USD, so gas in USD is technically better

Reply:
oh hmm i see this just converts it to a `CurrencyAmount`, wondering if you can avoid doing conversions at all here